mm.c

```diff
@@ -381,10 +380,24 @@ int mm_init(void)
 {
     // Initialize the free list
     for (int i = 0; i <= MAX_POWER; i++) {
-				SET_FREE_LIST_PTR(i, NULL);
+	    SET_FREE_LIST_PTR(i, NULL);
     }
 
+    char *heap_start; // Pointer to begining of heap 
+
+    if ((long)(heap_start = mem_sbrk(WORD_SIZE)) == -1)
+        return -1;
+
+    PUT_WORD(heap_start + WORD_SIZE, PACK(D_WORD_SIZE, TAKEN)); // Prolog header
```

Comment:
Shouldn't add `WORD_SIZE` because we are starting at the very beginning of the heap.
Should make size 0 because we don't define size like the book does.

Just use: `PUT_WORD(heap_start, PACK(0, TAKEN))`
and: `PUT_WORD(FTRP(heap_start), PACK(0, TAKEN))`